contracts/liquidityMining/KyberRewardLocker.sol

```diff
@@ -94,15 +94,31 @@ contract KyberRewardLocker is IKyberRewardLocker, PermissionAdmin {
     address account,
     uint256 quantity,
     uint256 startBlock
-  ) public override onlyRewardsContract(token) {
+  ) public payable override onlyRewardsContract(token) {
     require(quantity > 0, 'Quantity cannot be zero');
 
-    // transfer token from reward contract to lock contract
-    token.safeTransferFrom(msg.sender, address(this), quantity);
+    if (token == IERC20Ext(0)) {
+      require(msg.value == quantity, 'Invalid locked quantity');
+    } else {
+      // transfer token from reward contract to lock contract
+      token.safeTransferFrom(msg.sender, address(this), quantity);
+    }
 
     VestingSchedules storage schedules = accountVestingSchedules[account][token];
     uint256 schedulesLength = schedules.length;
     uint256 endBlock = startBlock.add(vestingDurationPerToken[token]);
+
+    // combine with the last schedule if they have the same start & end blocks
+    if (schedulesLength > 0) {
+      VestingSchedule storage lastSchedule = schedules.data[schedulesLength - 1];
+      if (lastSchedule.startBlock == startBlock && lastSchedule.endBlock == endBlock) {
+        lastSchedule.quantity = uint256(lastSchedule.quantity).add(quantity).toUint128();
+        accountEscrowedBalance[account][token] = accountEscrowedBalance[account][token].add(quantity);
+        emit VestingEntryCreated(token, account, startBlock, endBlock, quantity, schedulesLength - 1);
```

Comment:
This can be changed to `VestingEntryQueued`

Reply:
added new event, will need to state it clearly when reading data from events